pkg/diagnose/doc.go

```diff
@@ -0,0 +1,8 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// Package diagnose provides a way for other packages to register some diagnosis
+// that can be run tu debug connectivity issues
+package diagnose
```

Comment:
I think we need a package `README.md` focussed on diag writers: "I am currently implementing a new package, why should I add a diag and how should I do it? What are the requirements / good practices?"